okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -1108,14 +1092,13 @@ private void authForSystemInterface(ModuleDescriptor authMod, RoutingEntry filt,
     ModuleInstance authInst = new ModuleInstance(authMod, filt, inst.getPath(), HttpMethod.HEAD, inst.isHandler());
     doCallSystemInterface(headers, tenantId, null, authInst, modPerms, "", res -> {
       if (res.failed()) {
-        logger.warn("Auth check for systemInterface failed!");
-        fut.handle(new Failure<>(res.getType(), res.cause()));
+        fut.handle(res);
         return;
       }
       OkapiClient cli = res.result();
       String deftok = cli.getRespHeaders().get(XOkapiHeaders.TOKEN);
-      logger.debug("authForSystemInterface:"
-        + Json.encode(cli.getRespHeaders().entries()));
+      logger.debug("authForSystemInterface: {}",
+        Json.encode(cli.getRespHeaders().entries()));
```

Comment:
```suggestion
        () -> Json.encode(cli.getRespHeaders().entries()));
```
Avoid encoding when debug logging is diabled.